examples/datawarehouse/index.ts

```diff
@@ -44,19 +49,116 @@ const genericTableArgs: StreamingInputTableArgs = {
     scheduleExpression
 }
 
+// create two tables with kinesis input streams, writing data into hourly partitions in S3. 
 const dataWarehouse = new ServerlessDataWarehouse("analytics_dw", { isDev })
     .withStreamingInputTable(impressionsTableName, genericTableArgs)
     .withStreamingInputTable(clicksTableName, genericTableArgs);
 
 const impressionsInputStream = dataWarehouse.getInputStream(impressionsTableName);
 const clicksInputStream = dataWarehouse.getInputStream(clicksTableName);
 
-createEventGenerator("impression", impressionsInputStream.name);
-createEventGenerator("clicks", clicksInputStream.name);
 
+// Export a batch of outputs from the first two tables.
 export const impressionInputStream = impressionsInputStream.name;
 export const clickInputStream = clicksInputStream.name;
 export const databaseName = dataWarehouse.database.name;
 export const impressionTableName = dataWarehouse.getTable(impressionsTableName).name;
 export const clickTableName = dataWarehouse.getTable(clicksTableName).name;
 export const athenaResultsBucket = dataWarehouse.queryResultsBucket.bucket;
+
+const dwBucket = dataWarehouse.dataWarehouseBucket.bucket
+
+// Configure batch input table 'aggregates'
+const aggregateTableName = "aggregates";
+
+const aggregateTableColumns = [
+    {
+        name: "event_type",
+        type: "string"
+    },
+    {
+        name: "count",
+        type: "int"
+    },
+    {
+        name: "time",
+        type: "string"
+    }
+];
+
+const aggregationFunction = async (event: EventRuleEvent) => {
```

Comment:
How does lambda know the dependencies of this code? 
(moment, S3, etc.)

Reply:
Pulumi ultimately creates a closure serialization. More detail on this [here](https://www.pulumi.com/docs/tutorials/aws/serializing-functions/) if you're curious.

Reply:
Here's a gist that shows the serialized function that the lambda runs https://gist.github.com/EvanBoyle/a9584a5a923d96cb84bca9ae7a7ca7c4